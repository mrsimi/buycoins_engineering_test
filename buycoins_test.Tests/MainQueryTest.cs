
using buycoins_test.Queries;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json.Linq;
using Snapshooter.Xunit;
using System;
using System.Threading.Tasks;
using Xunit;

namespace buycoins_test.Tests
{
    public class MainQueryTest
    {
        [Fact]
        public async Task Test1()
        {
            var executor = await new ServiceCollection()
                    .AddGraphQLServer()
                    .AddQueryType<MainQuery>()
                    .BuildRequestExecutorAsync();

            var query = QueryRequestBuilder.New()
                    .SetQuery(" query{ userBankInfo { accountName} }")
                    .Create();

            IExecutionResult result = await executor.ExecuteAsync(query);

            //assert
            Assert.Null(result.Errors);
            var jsonResult = JObject.Parse(result.ToJson());

            var accountName = jsonResult["data"]["userBankInfo"]["accountName"]?.ToString();


            result.MatchSnapshot();
        }

        [Fact]
        public async Task GetUserAccountName_ReturnsOk()
        {

            var executor = await new ServiceCollection()
                    .AddGraphQLServer()
                    .AddQueryType<MainQuery>()
                    .BuildRequestExecutorAsync();

            var query = QueryRequestBuilder.New()
                    .SetQuery("query{ userAccountName(accountNumber: \"0161023149\", bankCode: \"058\") { accountName statusCode } }")
                    .Create();


            var result = await executor.ExecuteAsync(query);
            Assert.Null(result.Errors);

            var jsonResult = JObject.Parse(result.ToJson());

            var accountName = jsonResult["data"]["userAccountName"]["accountName"]?.ToString();
            int statusCode = 0;
            int.TryParse(jsonResult["data"]["userAccountName"]["statusCode"]?.ToString(), out statusCode);
            
            Assert.Equal(200, statusCode);
            result.MatchSnapshot();
        }
    }
}