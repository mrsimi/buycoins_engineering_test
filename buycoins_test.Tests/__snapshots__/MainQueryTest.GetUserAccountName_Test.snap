{
  "Label": null,
  "Path": null,
  "Data": {
    "userAccountName": {
      "accountName": "Adegoke Similoluwa Matthias"
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
